api/src/main/java/org/apache/iceberg/metrics/DefaultCounter.java

```diff
@@ -38,6 +38,11 @@ public void increment(long amount) {}
         public long value() {
           throw new UnsupportedOperationException("NOOP counter has no value");
         }
+
+        @Override
+        public String toString() {
+          return "NOOP counter";
```

Comment:
the default implementation of `toString()` would always use `value()` and show the `UnsupportedOperationException` in debugging sessions, thus we're just overriding this for Counter/Timer here